src/psyclone/psyir/nodes/container.py

```diff
@@ -187,6 +188,37 @@ def get_routine_psyir(self, name):
                 return routine
         return None
 
+    def resolve_routine(self, name):
+        '''This function returns a list of function names that might be
+        actually called when the routine `name` is called. In most cases
+        this is exactly name, but in case of a generic subroutine the
+        name might change. For now (since we cannot compare routine
+        signatures yet), we return the list of all possible functions that
+        might be called.
+
+        :param str name: the name of the routine to resolve
+
+        :returns: the names of those routines that may actually be invoked
+            when the routine `name` is called or an empty list if there is no
+            routine with that name in this container.
+        :rtype: list[None | psyclone.psyir.nodes.Routine]
```

Comment:
I believe that should be `rtype: list[str]`

Reply:
Oops again.